src/llama_stack_provider_ragas/remote/kubeflow/components.py

```diff
@@ -7,7 +7,51 @@
 load_dotenv()
 
 
-@dsl.component(base_image=os.environ["KUBEFLOW_BASE_IMAGE"])
+def get_base_image() -> str:
+    """Get base image from env, fallback to k8s ConfigMap, fallback to default image."""
+
+    base_image = os.environ.get("KUBEFLOW_BASE_IMAGE")
+    if base_image:
+        return base_image
+
+    try:
+        import logging
+
+        from kubernetes import client, config
+
+        from llama_stack_provider_ragas.constants import (
+            DEFAULT_RAGAS_PROVIDER_IMAGE,
+            RAGAS_PROVIDER_IMAGE_CONFIGMAP_KEY,
+            RAGAS_PROVIDER_IMAGE_CONFIGMAP_NAME,
+        )
+
+        logger = logging.getLogger(__name__)
+        logger.setLevel(logging.INFO)
+
+        try:
+            config.load_incluster_config()
+        except config.ConfigException:
+            config.load_kube_config()
+
+        api = client.CoreV1Api()
+
+        # For now, use hardcoded namespace for redhat-ods-applications
+        namespace = "redhat-ods-applications"
```

Comment:
To ensure this works with ODH, I would try to find the ConfigMap first on `redhat-ods-applications` (as it is now), then try `opendatahub`. If none has the CM, then throw the exception.